2. Determine the number of rows and columns in the dataset.

In [1]:
# importing pandas library for data manipulation
import pandas as pd

# reading the CSV file into a DataFrame
df = pd.read_csv("Airplane_Crashes_and_Fatalities_Since_1908.csv")

# getting the number of rows and columns in the DataFrame
number_of_rows, number_of_columns = df.shape
print(f"Number of rows: {number_of_rows}")
print(f"Number of columns: {number_of_columns}")

Number of rows: 5268
Number of columns: 13


3. Display the last 75 rows in the dataset.

In [2]:
# getting the last 75 rows of the DataFrame
last_75_rows = df.tail(75)
# displaying the last 75 rows of the DataFrame
last_75_rows

,Date,Time,Location,Operator,Flight #,Route,Type,Registration,cn/In,Aboard,Fatalities,Ground,Summary
5193,03/15/2008,08:15,"Nbagu, Nigeria",Trade Wings Aviation Ltd.,NaN,Lagos - Bebi Air Strip,Beechcraft 1900D,5N-JAH,UE-322,3.0,3.0,0.0,The plane crashed while en route. Wreckage was...
5194,04/03/2008,11:00,"Lawa, Suriname",Blue Wing Airlines,NaN,Paramaribo - Lawa,Antonov An-28,PZ-TSO,1AJ007-17,19.0,19.0,0.0,While attempting to land the crew aborted the ...
5195,04/09/2008,23:27,"Bundeena, Australia",Avtex Air Services,NaN,Sydney - Brisbane,Swearingen SA227AC Metroliner III,VH-OZA,AC-600,1.0,1.0,0.0,The pilot of the mail plane reported some mino...
5196,04/11/2008,22:15,"Chrisinau, Moldova",Kata Transportation,NaN,"Chrisinau, Moldova - Antalya, Turkey - Sudan",Antonov An-32,ST-AZL,3009,8.0,8.0,0.0,Just minutes after the take off the plane trie...
5197,04/15/2008,14:30,"Goma, Congo",Hewa Bora Airways,NaN,Goma - Kisangani - Kinshasa,McDonnell Douglas DC-9-51,9Q-CHN,47731,85.0,0.0,47.0,"While attempting to takeoff, the plane failed ..."
...,...,...,...,...,...,...,...,...,...,...,...,...,...
5263,05/20/2009,06:30,"Near Madiun, Indonesia",Military - Indonesian Air Force,NaN,Jakarta - Maduin,Lockheed C-130 Hercules,A-1325,1982,112.0,98.0,2.0,"While on approach, the military transport cras..."
5264,05/26/2009,NaN,"Near Isiro, DemocratiRepubliCongo",Service Air,NaN,Goma - Isiro,Antonov An-26,9Q-CSA,5005,4.0,4.0,NaN,The cargo plane crashed while on approach to I...
5265,06/01/2009,00:15,"AtlantiOcean, 570 miles northeast of Natal, Br...",Air France,447,Rio de Janeiro - Paris,Airbus A330-203,F-GZCP,660,228.0,228.0,0.0,The Airbus went missing over the AtlantiOcean ...
5266,06/07/2009,08:30,"Near Port Hope Simpson, Newfoundland, Canada",Strait Air,NaN,Lourdes de BlanSablon - Port Hope Simpson,Britten-Norman BN-2A-27 Islander,C-FJJR,424,1.0,1.0,0.0,The air ambulance crashed into hills while att...


4. Each column except the first one has some missing data. For each column, name
the most appropriate method for treating the missing data. Justify your choices.

In [3]:
# getting the number of missing values in each column of the DataFrame
df.isnull().sum()

# displaying the summary information of the DataFrame
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5268 entries, 0 to 5267
Data columns (total 13 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Date          5268 non-null   object 
 1   Time          3049 non-null   object 
 2   Location      5248 non-null   object 
 3   Operator      5250 non-null   object 
 4   Flight #      1069 non-null   object 
 5   Route         3561 non-null   object 
 6   Type          5241 non-null   object 
 7   Registration  4933 non-null   object 
 8   cn/In         4040 non-null   object 
 9   Aboard        5246 non-null   float64
 10  Fatalities    5256 non-null   float64
 11  Ground        5246 non-null   float64
 12  Summary       4878 non-null   object 
dtypes: float64(3), object(10)
memory usage: 535.2+ KB


1. Date - The df.info() output shows Date has 5268 non-null values out of 5268 total rows, which means that there are no missing values in this column, so no treatment is needed.

2. Time - The df.info() output shows Time has 3049 non-null values out of 5268 total rows, a big portion missing from this column. Since time is text information and missing times can't be determined correctly or inferred from the other columns, these missing values are replaced with "Unknown."

3. Location - The df.info() output shows Location has 5248 non-null values out of 5268 total rows, which means very few rows are missing in this field. Since Location is categorical and is required for later analysis, these few rows are dropped rather than imputing them. 

4. Operator - The df.info() output shows Operator has 5250 non-null values out of 5268 total rows, which means that very few rows are missing this field. Since Operator is categorical information and can't be determined, these rows are dropped rather than filled in.

5. Flight # - The df.info() output shows Flight # has only 1069 non-null values out of 5268 total rows, which means that most of this column is missing. Since Flight # is a unique identifier and can't be determined from the other columns, it cannot be imputed reliably, and dropping the rows would remove most of the dataset, so the column is dropped entirely.

6. Route - The df.info() output shows Route has 3561 non-null values out of 5268 total rows, which means that there is a lot of data missing from this column. Since Route is text information and can't be reliably calculated or inferred from the other columns, these missing values are replaced with "Unknown."

7. Type - The df.info() output shows Type has 5241 non-null values out of 5268 total rows, which means very few rows are missing this field. Since Type is categorical information and can't be reliably guessed, these rows are dropped rather than filled.

8. Registration - The df.info() output shows Registration has 4933 non-null values out of 5268 total rows, meaning a small portion of this column is missing. Since Registration is categorical information and can't be reliably guessed, these rows are dropped rather than filling them in.

9. cn/In - The df.info() output shows cn/In has 4040 non-null values out of 5268 total rows, which means that there is a fair portion of this column is missing. Since cn/In is an identifier and can't be reliably calculated or inferred from the other columns, these missing values are replaced with "Unknown."

10. Aboard - The df.info() output shows Aboard has 5246 non-null values out of 5268 total rows, which means that very few rows are missing this field. Since Aboard is a numeric count that is likely skewed (a small number of large flights compared to many smaller ones), missing values are filled with the median, because it is less affected by outliers than the mean.

11. Fatalities - The df.info() output shows Fatalities has 5256 non-null values out of 5268 total rows, which means that very few rows are missing this field. Since Fatalities is a numeric count that is likely skewed (a small number of crashes with very high fatalities compared to many smaller ones), missing values are filled with the median, because it is less affected by outliers than the mean.

12. Ground - The df.info() output shows Ground has 5246 non-null values out of 5268 total rows, meaning very few rows are missing this field. Since Ground is a numeric count, missing values are filled with the median, which is more robust to outliers than the mean and is consistent with the treatment applied to Aboard and Fatalities.

13. Summary - The df.info() output shows Summary has 4878 non-null values out of 5268 total rows, which means a portion of this column is missing. Since Summary is text information and can't be reliably calculated or inferred from the other columns, these missing values are replaced with "Unknown."